pr_watch/tests/test_tasks.py

```diff
@@ -141,3 +142,134 @@ def test_watch_pr_rate_limit_exceeded(
+        ]
+
+        def fake_pr_list_from_usernames(username_list, fork_name):
+            """
+            Simulate a GitHub answer to avoid doing API calls.
+            When asked about PRs in the first repository, answer the first test PR,
+            and when asked about the second, answer the second one.
+            """
+            if fork_name == 'edx/edx-platform':
+                return []
+            elif fork_name == 'source/repo1':
+                return [test_data[0]['pr']]  # first PR
+            elif fork_name == 'source/repo2':
+                return [test_data[1]['pr']]  # second PR
+            else:
+                raise NotImplementedError()
+
+        # Substitute GitHub API calls by our results
+        mock_get_username_list.return_value = ['itsjeyd']
+        mock_get_pr_list_from_usernames.side_effect = fake_pr_list_from_usernames
+        # Commit ID is not used in this test, so we're using the same in both PRs
+        mock_get_commit_id_from_ref.return_value = '7' * 40
+
+        self.assertEqual(mock_spawn_appserver.call_count, 0)
+
+        # Check github (with our mocked response), this will detect 2 watched PRs and spawn 1 instance for each
+        tasks.watch_pr()
+
+        self.assertEqual(mock_spawn_appserver.call_count, 2)
+
+        # Now do checks twice, once for every instance, checking the appropriate values each time
+        for pr_number in [1, 2]:
+            # "pr" contains the data we expect to see: URL, settings, and the PR object itself
+            pr = test_data[pr_number - 1]
+
+            new_instance_ref_id = mock_spawn_appserver.mock_calls[pr_number - 1][1][0]
+            instance = OpenEdXInstance.objects.get(ref_set__pk=new_instance_ref_id)
+            subdomain_part = 'pr2300{}'.format(pr_number) # e.g. pr23001, pr23002, etc.
+            self.assertEqual(instance.internal_lms_domain, '{}.sandbox.awesome.hosting.org'.format(subdomain_part))
+            self.assertEqual(instance.internal_lms_preview_domain,
+                             'preview-{}.sandbox.awesome.hosting.org'.format(subdomain_part))
+            self.assertEqual(instance.internal_studio_domain,
+                             'studio-{}.sandbox.awesome.hosting.org'.format(subdomain_part))
+            self.assertEqual(instance.edx_platform_repository_url, 'https://github.com/fork/repo.git')
+            self.assertEqual(instance.edx_platform_commit, '7' * 40)
+            self.assertEqual(instance.openedx_release, 'master')
+            self.assertEqual(yaml.load(instance.configuration_extra_settings), pr['expected_settings'])
+            # Configuration settings from WatchedFork have precedence over the ones from PR
+            self.assertEqual(instance.configuration_source_repo_url,
+                             'https://github.com/open-craft/configuration-fromwatchedfork')
+            self.assertEqual(instance.configuration_version, 'named-release/elder-fromwatchedfork')
```

Comment:
As [mentioned above](https://github.com/open-craft/opencraft/pull/246/files#diff-4ccb9a3bd803e3e653eac6d28f3d5be4R114), I think the PR settings should take precedence over the `WatchedFork` settings.  So this would need to be changed to:
```python
           self.assertEqual(instance.configuration_source_repo_url,
                            'https://github.com/open-craft/configuration')
           self.assertEqual(instance.configuration_version, 'named-release/elder')
```